<?xml version="1.0" encoding="UTF-8"?>

<launch>
        <arg name="run_simulation" default="true"/>
        <!--arguments passed to my_simulation/my_simulation.launch -->
	<arg name="world"  default="small_office_inventory"/>
	<arg name="gzpose" default="-x 0.0 -y 0.0 -z 0.0 -R 0.0 -P 0.0 -Y 0.0"/>
        <arg name="spawn_objects" default="true"/>

	<node pkg="path_finding" type="turn_towards.py" name="turn_towards_server"/>

	<node pkg="pickup" type="pickup_objects.py" name="pickup_objects" output="screen" required="true"/>
	<rosparam file="$(find my_simulation)/params/$(arg world).yaml" />

	<include if="$(arg run_simulation)" file="$(find my_simulation)/launch/my_simulation.launch" >
		<arg name="world"  value="$(arg world)" />
		<arg name="gzpose" value="$(arg gzpose)" />
		<arg name="global_planner" value="global_planner"/>
                <arg name="spawn_objects" value="$(arg spawn_objects)"/>
	</include>

	<node pkg="items_locating" type="find_object.py" name="find_object"/>
	<node pkg="path_finding" type="plan_path.py" name="plan_path" output="screen"/>
</launch>

